x-pack/solutions/security/plugins/security_solution/common/siem_migrations/common/resources/index.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SiemMigrationResourceBase, SiemMigrationResourceData } from '../../model/common.gen';
+import { splResourceIdentifier } from '../../rules/resources/splunk/splunk_identifier';
+import type { VendorResourceIdentifier } from '../../rules/resources/types';
+import type { SiemMigrationVendor } from '../../types';
+
+const identifiers: Record<SiemMigrationVendor, VendorResourceIdentifier> = {
+  splunk: splResourceIdentifier,
+};
+
+export abstract class ResourceIdentifier<I> {
+  protected identifier: VendorResourceIdentifier;
+
+  constructor(protected readonly vendor: SiemMigrationVendor) {
+    // The constructor may need query_language as an argument for other vendors
+    this.identifier = identifiers[this.vendor];
+  }
+
+  public abstract fromOriginal(
+    item?: I
+  ): SiemMigrationResourceBase[] | Promise<SiemMigrationResourceBase[]>;
```

Comment:
Is there any reason to accept both return types? Wouldn't it make more sense to always expect a promise?

Reply:
Okay, but no need to convert to a promise explicitly, if the function is `async`, the return is always converted to a promise automatically. It would look like this:

```ts
public async fromOriginal(originalRule: OriginalRule): Promise<RuleMigrationResourceBase[]> {
  return this.identifier(originalRule.query);
}
```

but it's fine.

Reply:
Oh right.. i did the change. Thanks.